app/controllers/users/base.rb

```diff
@@ -15,5 +16,10 @@ def business_present_access
     def business_nil_access
       redirect_to new_users_business_path, flash: { danger: '事業所を登録して下さい' } if current_user.business.nil?
     end
+
+    # current_userに紐づく事業所を取得
+    def set_business
```

Comment:
```suggestion
    def current_business
       @business ||= current_user&.business
    end
```

上記のようにbaseコントローラーに定義して、各コントローラーで使うのがいいですね👍current_userみたいに使えます。
参考：https://qiita.com/ryosuketter/items/3e5f7cc46163a8c247ed

Reply:
この場合だとユーザーがログインしていなかったり、ユーザーにbusinessがない場合はnilが返ることになります。
また管理ユーザーと一般ユーザーの場合も考慮したいですね。

```suggestion
    if current_userが管理ユーザーの場合
      @business ||= current_user&.business
    elsif curren_userが一般ユーザーの場合
      @business ||= current_user&.parent&.business
    end
```